bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/AbstractResultTab.java

```diff
@@ -0,0 +1,225 @@
+		Display display = Display.getDefault();
+
+		boolean successOfAutomaticTests = this.calculateSuccessOfAutomaticTests(feedbacks);
+		double points = this.calculatePoints(currentExercise, feedbacks);
+
+		double score = this.calculateScore(currentExercise, points);
+
+		String title = currentExercise == null ? "Unknown Task" : currentExercise.getTitle();
+
+		this.btnResultSuccessful.setForeground(successOfAutomaticTests ? display.getSystemColor(SWT.COLOR_GREEN) : display.getSystemColor(SWT.COLOR_RED));
+		this.btnResultSuccessful.setText(successOfAutomaticTests ? "Test(s) succeeded" : "Test(s) failed");
+
+		this.lblResultExerciseShortName.setText(title);
+		this.lblResultExerciseDescription.setText(completionTime == null ? "" : completionTime);
+		this.resultScore.setText(resultString == null ? String.format(Locale.ENGLISH, "Score: %.2f%%", score) : resultString);
+		this.lblPoints.setText(String.format(Locale.ENGLISH, "Points: %.2f", points));
+		this.layout();
+
+	}
+
+	private double calculatePoints(IExercise exercise, List<Feedback> feedbacks) {
+		if (exercise == null || feedbacks == null) {
+			return Double.NaN;
+		}
+		return feedbacks.stream().mapToDouble(f -> f.getCredits() == null ? 0.0 : f.getCredits()).sum();
+	}
+
+	private double calculateScore(IExercise exercise, double points) {
+		if (exercise == null || Double.isNaN(points)) {
+			return Double.NaN;
+		}
+
+		points = Math.max(0, Math.min(exercise.getMaxPoints(), points));
+		return points / exercise.getMaxPoints() * 100;
+	}
+
+	private boolean calculateSuccessOfAutomaticTests(List<Feedback> feedbacks) {
+		if (feedbacks == null || feedbacks.isEmpty()) {
+			return false;
+		}
+		return feedbacks.stream().filter(f -> f.getFeedbackType() == FeedbackType.AUTOMATIC).allMatch(f -> Boolean.TRUE.equals(f.getPositive()));
+	}
+
+	private void addFeedbackToTable(Table table, List<Feedback> entries) {
+		if (entries == null) {
+			return;
+		}
+
+		entries.stream().sorted().forEach(feedback -> {
+			var name = feedback.getFeedbackType() != FeedbackType.AUTOMATIC && feedback.getText() == null ? "Tutor Comment" : feedback.getText();
+			this.createTableItemsForFeedback(table, name, feedback);
+		});
+
+	}
+
+	private void createTableItemsForFeedback(Table table, String name, Feedback feedback) {
+		String roundedCredits = feedback.getCredits() == null ? "0.00" : String.format(Locale.ENGLISH, "%.2f", feedback.getCredits());
+		String success = this.calculateSuccessMessage(feedback);
+		int colorIndicator = this.calculateSuccessColorIndicator(feedback);
+
+		final TableItem item = new TableItem(table, SWT.NULL);
+		item.setData(feedback);
+		item.setText(0, name);
+		item.setText(1, "" + roundedCredits);
+		item.setText(2, success);
+		item.setForeground(2, Display.getDefault().getSystemColor(colorIndicator));
+		item.setText(3, feedback.getDetailText() != null ? CHECK_MARK_IN_UTF8 : X_MARK_IN_UTF8);
```

Comment:
I'm not a huge fan of the checkmark / cross indicator. Seems quite unintuitive to have a checkmark even if your test failed.

Reply:
Any suggestions? I'm very happy with other indicators :)

Reply:
Maybe just "present" or something like that. I'll think about it, it's not a big problem but feels just a bit counterintuitive.